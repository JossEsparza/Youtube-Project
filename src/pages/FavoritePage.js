import React from 'react';
import './FavoritePage.scss';

export const FavoritePage = () => {
  return (
    <div className="fav_container">
      <h1> This is the FavoritePage</h1>
    </div>
  );
};
