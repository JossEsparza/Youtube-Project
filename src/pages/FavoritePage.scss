.dark {
  .fav_container {
    background-color: #1b1d21;
    h1 {
      color: #cbcccd;
    }
  }
}

.fav_container {
  width: 100vw;
  height: 100vh;
}
