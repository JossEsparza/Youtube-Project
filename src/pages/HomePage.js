import React from 'react';
import VideoCard from '../components/cards/VideoCard';
// import { useFetch } from '../hooks/useFetch';
import './HomePage.scss';

export const HomePage = () => {
  const data = require('../mocks/youtube-videos-mock.json');
  // const key = 'AIzaSyABmgpPt9OIaG3TfqS9HlzmUv54cDMFUQQ';
  const error = 0;
  const loading = 0;
  // const { data, loading, error } = useFetch(
  //   'https://www.googleapis.com/youtube/v3/search?part=id,snippet&maxResults=15&q=tenis&key=' +
  //     key
  // );

  if (error) {
    return <p>{error.message}</p>;
  }

  const filtered_video = data?.items?.filter((video) => {
    return video.id.kind.includes('youtube#video');
  });

  return (
    <div className="info_container">
      <div className="video_container">
        {loading ? (
          <p>Loading!</p>
        ) : (
          filtered_video.map((video) => <VideoCard key={video.id.videoId} {...video} />)
        )}
      </div>
    </div>
  );
};
