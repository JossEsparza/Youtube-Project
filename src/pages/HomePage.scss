.video_container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.info_container {
  overflow: scroll;
  height: 100vh;
}

.dark {
  .video_container {
    background-color: #1b1d21;
  }
}
