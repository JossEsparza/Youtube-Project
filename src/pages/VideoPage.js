import React from 'react';
// import { useFetch } from '../hooks/useFetch';
// import { useLocation } from 'react-router-dom';
import RelatedVideos from '../components/related_videos/RelatedVideos';
import { GrChannel } from 'react-icons/gr';
import './VideoPage.scss';

const VideoPage = () => {
  const data = require('../mocks/youtube-specific-video-mock.json');
  const error = 0;
  const loading = 0;

  // const location = useLocation();

  // const { data, loading, error } = useFetch(
  //   'https://www.googleapis.com/youtube/v3/videos?part=id, snippet,player&key=' +
  //     process.env.REACT_APP_YOUTUBE_API_KEY +
  //     '&id=' +
  //     location.pathname.substring(1)
  // );

  const { items } = data;

  if (error) {
    return <p>{error.message}</p>;
  }

  return (
    <div className="video_page_container">
      {loading ? (
        <p>Loading!</p>
      ) : (
        <div className="video_page_info_container">
          <div className="video_player">
            <iframe
              title={items[0].id}
              width="100%"
              height="100%"
              src={`//www.youtube.com/embed/${items[0].id}`}
              frameBorder="0"
              allowFullScreen
              allow={
                'accelerometer; autoplay; clipboard-write;' +
                'encrypted-media; gyroscope; picture-in-picture'
              }
            />
            <h3>{items[0].snippet.title}</h3>
            <p className="channel_icon">
              <GrChannel /> {items[0].snippet.channelTitle}
            </p>
            <span>Description: {items[0].snippet.description}</span>
          </div>
          <RelatedVideos />
        </div>
      )}
    </div>
  );
};

export default VideoPage;
