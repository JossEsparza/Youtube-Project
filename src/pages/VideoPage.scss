.video_page_info_container {
  display: flex;
  flex-direction: row;
  margin-top: 20px;
}

.video_player {
  justify-items: center;
  margin-left: 40px;
  margin-right: 40px;
}

p {
  font-weight: bold;
  margin-bottom: -1px;
}

.video_page_container {
  height: 100vh;
}

.dark {
  h4 {
    color: #cbcccd;
  }
  p {
    color: #a3a3a6;
  }

  .video_page_container {
    background-color: #1b1d21;
  }
}
