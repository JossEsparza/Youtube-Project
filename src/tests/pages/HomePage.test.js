import React from 'react';
import { mount } from 'enzyme';
import { HomePage } from '../../pages/HomePage';
import { ToggleContext } from '../../state/ToggleContext';

describe('Test in <Home Page />', () => {
  test('should be displayed correctly', () => {
    const globalReducer = {
      inputValue: 'changuitos',
      currentToggle: false,
      isEnabled: false,
    };
    const wrapper = mount(
      <ToggleContext.Provider
        value={{
          globalReducer,
        }}
      >
        <HomePage />
      </ToggleContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
