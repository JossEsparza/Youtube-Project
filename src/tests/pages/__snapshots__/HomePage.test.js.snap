// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Home Page /> should be displayed correctly 1`] = `ReactWrapper {}`;
